verde/utils.py

```diff
@@ -681,6 +683,37 @@ def kdtree(coordinates, use_pykdtree=True, **kwargs):
     return tree
 
 
+def fill_nans(grid):
+    """
+    This methos is responsible for fill the NaN values in the grid using the KNN algorithm.
+
+    Parameters
+    ----------
+    grid : :class:`xarray.DataArray`
+        A 2D grid with one or more data variables.
+    Returns
+    -------
+    grid : :class:`xarray.DataArray`
+        A 2D grid with the NaN values filled.
+    """
+
+    filled_grid = grid.copy()
+
+    not_nan_values = np.argwhere(~np.isnan(grid.values))
+    unknown_indices = np.argwhere(np.isnan(grid.values))
+
+    knn_imputer = vd.KNeighbors()
+    easting, northing = not_nan_values[:, 0], not_nan_values[:, 1]
+    knn_imputer.fit((easting, northing), grid.values[not_nan_values[:, 0],
+                    not_nan_values[:, 1]])
+    predicted_values = knn_imputer.predict((easting, northing))
+
+    for i, idx in enumerate(unknown_indices):
+        filled_grid[tuple(idx)] = predicted_values[i]
```

Comment:
Instead of this, you could use `knn.grid` and pass in the coordinates of the original grid.